sdk/core/System.Net.ClientModel/api/System.Net.ClientModel.net6.0.cs

```diff
@@ -0,0 +1,407 @@
+        public static T? Read<T>(System.BinaryData data, System.Net.ClientModel.ModelReaderWriterOptions? options = null) where T : System.Net.ClientModel.Core.IModel<T> { throw null; }
+        public static System.BinaryData Write(object model, System.Net.ClientModel.ModelReaderWriterFormat format) { throw null; }
+        public static System.BinaryData Write(object model, System.Net.ClientModel.ModelReaderWriterOptions? options = null) { throw null; }
+        public static System.BinaryData Write<T>(T model, System.Net.ClientModel.ModelReaderWriterFormat format) where T : System.Net.ClientModel.Core.IModel<T> { throw null; }
+        public static System.BinaryData Write<T>(T model, System.Net.ClientModel.ModelReaderWriterOptions? options = null) where T : System.Net.ClientModel.Core.IModel<T> { throw null; }
+    }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct ModelReaderWriterFormat : System.IEquatable<System.Net.ClientModel.ModelReaderWriterFormat>
+    {
+        private readonly object _dummy;
+        private readonly int _dummyPrimitive;
+        public ModelReaderWriterFormat(string value) { throw null; }
+        public static System.Net.ClientModel.ModelReaderWriterFormat Json { get { throw null; } }
+        public static System.Net.ClientModel.ModelReaderWriterFormat Wire { get { throw null; } }
+        public static System.Net.ClientModel.ModelReaderWriterFormat Xml { get { throw null; } }
+        public bool Equals(System.Net.ClientModel.ModelReaderWriterFormat other) { throw null; }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override bool Equals(object? obj) { throw null; }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override int GetHashCode() { throw null; }
+        public static bool operator ==(System.Net.ClientModel.ModelReaderWriterFormat left, System.Net.ClientModel.ModelReaderWriterFormat right) { throw null; }
+        public static implicit operator System.Net.ClientModel.ModelReaderWriterFormat (string value) { throw null; }
+        public static bool operator !=(System.Net.ClientModel.ModelReaderWriterFormat left, System.Net.ClientModel.ModelReaderWriterFormat right) { throw null; }
+        public override string ToString() { throw null; }
+    }
+    public partial class ModelReaderWriterOptions
+    {
+        public ModelReaderWriterOptions() { }
+        public ModelReaderWriterOptions(System.Net.ClientModel.ModelReaderWriterFormat format) { }
+        public static System.Net.ClientModel.ModelReaderWriterOptions DefaultWireOptions { get { throw null; } }
+        public System.Net.ClientModel.ModelReaderWriterFormat Format { get { throw null; } }
+    }
+    public partial class NullableResult<T> : System.Net.ClientModel.Result
```

Comment:
If we make this the API for Result<T>, do we have a need for both it and a `NullableResult<T>`? In other words, if we start with Result<T> being nullable, is HasValue and nullability not enough to handle the scenarios for both types?